cli/src/cli_util.rs

```diff
@@ -2029,7 +2016,20 @@ pub fn print_dropped_signatures(ui: &Ui, num_dropped: usize) -> Result<(), std::
     }
     Ok(())
 }
- 
+
+pub fn print_rebase_info(
+    ui: &Ui,
+    rebase_counts: &RebaseCounts,
+    message: &str,
+) -> Result<(), std::io::Error> {
+    let num_rebased = rebase_counts.rebased;
+    if num_rebased > 0 {
+        let message = message.replace("{}", &format!("{}", num_rebased)) + "\n";
```

Comment:
nit: I personally avoid this kind of manual templating, and use this helper function only for the `"Rebased {} descendant commits"` case.

Reply:
This was done to address your comment here: https://github.com/martinvonz/jj/pull/3007#discussion_r1489365305

Maybe I misunderstood you? I'm not sure what you were intending then

Reply:
I just meant the common case can be extracted to the helper function.

```rust
fn print_rebase_info(
    ui: &Ui,
    rebase_counts: &RebaseCounts,
) -> Result<(), std::io::Error> {
    let num_rebased = rebase_counts.rebased;
    if num_rebased > 0 {
        writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits");
...
```

Reply:
Ok, but there is a lot of common logic that has been moved into this fn where the only changing component is the message suffix. Does it not make sense to have a util like this regardless?

Reply:
I tend to avoid reinventing string formatting function because it can get messier (by adding more parameters, formatting specifier, etc.), and I though deduplicating the common `"Rebased {num_rebased} descendant commits"` message worth the effort. That's just my preference, though.

```
% rg 'Rebased \{'
cli/src/cli_util.rs:729:                                "Rebased {num_rebased} descendant commits onto commits rewritten \
cli/src/cli_util.rs:963:                "Rebased {num_rebased} descendant commits off of commits rewritten from git"
cli/src/cli_util.rs:1427:                    "Rebased {num_rebased} descendant commits onto updated working copy"
cli/src/cli_util.rs:1491:            writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
cli/src/commands/new.rs:207:        writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
cli/src/commands/diffedit.rs:109:            writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
cli/src/commands/restore.rs:118:            writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
cli/src/commands/split.rs:153:        writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
cli/src/commands/untrack.rs:102:        writeln!(ui.stderr(), "Rebased {num_rebased} descendant commits")?;
cli/src/commands/rebase.rs:306:    writeln!(ui.stderr(), "Rebased {num_rebased} commits")?;
cli/src/commands/abandon.rs:77:            "Rebased {num_rebased} descendant commits onto parents of abandoned commits"
```